mne/time_frequency/psd.py

```diff
@@ -11,17 +11,19 @@
 from .multitaper import _psd_multitaper
 
 
-def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func):
+def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func, padding):
     """Aux function."""
     return func(epoch, fs=fs, nperseg=nfft, noverlap=noverlap,
-                nfft=nfft, window='hann')[2][..., freq_mask, :]
+                nfft=padding, window='hann')[2][..., freq_mask, :]
```

Comment:
@mmagnuski can you summarize the new logic? You add a padding parameter which can override the n_fft if it is not None?